### ISSUE TITLE
About the tT_loss

### ISSUE THREAD

**Issue by zzc681** [XiangLi1999/Diffusion-LM#63]
Hi, Thanks for your excellent work, but I have a small question about the loss function. When I was reading the code, I found that tT_loss calculates the loss between X_t and 0. Is there any meaning to doing this?
The code in the gaussian_diffusion.py, the function training_losses_e2e in class GaussianDiffusion
`out_mean, _, _ = self.q_mean_variance(x_start, torch.LongTensor([self.num_timesteps - 1]).to(x_start.device))`
`tT_loss =  mean_flat(out_mean ** 2)`

**Comment by RyuKT** [comment]
Same question for me.

The other loss terms written in training_losses_e2e are clear, which are also described in the paper (Equation 2). But I don't quite understand this tT_loss: why loss is calculated on each timestep of this forward diffusion process?

**Comment by ryuliuxiaodong** [comment]
Same question for me.

The other loss terms written in training_losses_e2e are clear, which are also described in the paper (Equation 2). But I don't quite understand this tT_loss: why loss is calculated on each timestep of this forward diffusion process?

**Comment by K0ntact** [comment]
Seems like `tT_loss` is to calculate how well the input are diffused into Normal Gaussian noise, since it is calculated by diffusing `x_start` to the last diffusion step via `q_mean_variance`.

https://github.com/XiangLi1999/Diffusion-LM/blob/759889d58ef38e2eed41a8c34db8032e072826f4/improved-diffusion/improved_diffusion/gaussian_diffusion.py#L252-L259

 The squared difference between `out_mean` and normal distribution mean (0) is `tT_loss`, which explains why it is written as `out_mean ** 2`.

https://github.com/XiangLi1999/Diffusion-LM/blob/759889d58ef38e2eed41a8c34db8032e072826f4/improved-diffusion/improved_diffusion/gaussian_diffusion.py#L1566-L1567
